bin/src/daemon.rs

```diff
@@ -113,6 +123,61 @@ fn run_ctrl_c_thread(
     })
 }
 
+async fn update_ui(task_tracker: &TaskTracker,  config: &config::Config, mut ui_shutdown_rx: oneshot::Receiver<()>){
+    static IMAGE_DIR: Dir<'_> = include_dir!("$CARGO_MANIFEST_DIR/static/images/");
+    let display_level = config.ui_level;
+    if display_level == 0 {
+        info!("Invisible mode, not spawning UI.");
+    }
+
+    task_tracker.spawn_blocking(move || {
+        let mut fb: Framebuffer = Framebuffer::new();
+        // this feels wrong, is there a more rusty way to do this?
+        let img;
+        if display_level == 2 {
+            img = IMAGE_DIR.get_file("orca.gif").unwrap().contents();
+        } else if display_level == 3{
+            img = IMAGE_DIR.get_file("eff.png").unwrap().contents();
+        } else {
+            img = &[];
+        }
+        loop {
+            match ui_shutdown_rx.try_recv() {
+                Ok(_) => {
+                    info!("received UI shutdown");
+                    break;
+                },
+                Err(TryRecvError::Empty) => {},
+                Err(e) => panic!("error receiving shutdown message: {e}")
+            
+            }
+            match display_level  {
+                1 => {
+                    fb.draw_line(framebuffer::Color565::Green, 2);
+                },
+                2 => {
+                    fb.draw_gif(img);
+                },
+                3 => {
+                    fb.draw_img(img)
+                }
```

Comment:
```suggestion
                2 | 3 => {
                    fb.draw_gif(img.unwrap());
                },
```

Reply:
Nope these are separate functions, I have split out draw gif and draw image

Reply:
also we arleady unwrap and return a result in 138/140 no?